tests/check_source_tests.py

```diff
@@ -69,7 +69,59 @@ def test_devel_project(self):
         """Accepts a request coming from a devel project"""
         self._setup_devel_project()
 
-        req_id = self.wf.create_submit_request(SRC_PROJECT, 'blowfish').reqid
+        req_id = self.wf.create_submit_request(SRC_PROJECT, 'blowfish', add_commit = False).reqid
+
+        self.assertReview(req_id, by_user=(self.bot_user, 'new'))
+
+        self.review_bot.set_request_ids([req_id])
+        self.review_bot.check_requests()
+
+        self.assertReview(req_id, by_user=(self.bot_user, 'accepted'))
+        self.assertReview(req_id, by_group=(REVIEW_TEAM, 'new'))
+
+    def test_missing_patch_in_changelog(self):
+        """Reject a request if it adds patch and it is not mentioned in changelog"""
+        # devel files contain patch but not changes
+        self._setup_devel_project(devel_files='blowfish-with-patch')
+
+        req_id = self.wf.create_submit_request(self.devel_package.project,
+            self.devel_package.name, add_commit = False).reqid
+
+        self.assertReview(req_id, by_user=(self.bot_user, 'new'))
+
+        self.review_bot.set_request_ids([req_id])
+        self.review_bot.check_requests()
+
+        review = self.assertReview(req_id, by_user=(self.bot_user, 'declined'))
+        self.assertIn(
+            'A patch (test.patch) is being added without this addition being mentioned in the changelog.',
+            review.comment
+        )
+
+    def test_patch_in_changelog(self):
+        """Accepts a request if it adds patch and it is mentioned in changelog"""
+        self._setup_devel_project()
+
+        req_id = self.wf.create_submit_request(self.devel_package.project,
+            self.devel_package.name, add_commit = False).reqid
+
+        self.assertReview(req_id, by_user=(self.bot_user, 'new'))
```

Comment:
Same again, you could save some lines if you wish. This basically duplicates the code of `assertReviewBot`

Reply:
yeah, I duplicate existing code to prevent another source if mistakes, but cleaning makes sense.